File: adaptors/jsaga-adaptor-batchssh/src/fr/in2p3/jsaga/adaptor/batchssh/job/BatchSSHJobStatus.java
package fr.in2p3.jsaga.adaptor.batchssh.job;

import fr.in2p3.jsaga.adaptor.job.SubState;
import fr.in2p3.jsaga.adaptor.job.monitor.JobStatus;

/* ***************************************************
* File:   BatchSSHJobStatus
* Author: Yassine BACHAR & Taha BENYEZZA
* Date:   12 Fevrier 2011
* ***************************************************/

public class BatchSSHJobStatus extends JobStatus {
   
	private int m_retCode;
	
    public BatchSSHJobStatus(String jobId, SubState status) {
        super(jobId,status, status.toString());
    }

    public BatchSSHJobStatus(String jobId, String state, int returnCode) {
    	super(jobId, state, state);
    	m_retCode = returnCode;
    }
    
    public BatchSSHJobStatus(String jobId, String state) {
    	super(jobId, state, state);
    }
    
    public String getModel() {
        return "pbs-ssh";
    }

    public SubState getSubState() {
    	if (m_nativeStateCode.equals("C") || m_nativeStateCode.equals("E")) {
    		if (m_retCode == 0) return SubState.DONE;
    		else if (m_retCode == 271) return SubState.CANCELED;
    		else return SubState.FAILED_ERROR;
    	} else if (m_nativeStateCode.equals("H")) {
            return SubState.SUSPENDED_ACTIVE;
    	} else if (m_nativeStateCode.equals("Q")) {
            return SubState.RUNNING_QUEUED;
    	} else if (m_nativeStateCode.equals("S") || m_nativeStateCode.equals("W")) {
            return SubState.SUSPENDED_QUEUED;
    	} else  { // "R"
            return SubState.RUNNING_ACTIVE;
    	} 
    }

}

File: adaptors/jsaga-adaptor-batchssh/src/fr/in2p3/jsaga/adaptor/batchssh/job/BatchSSHJobStatus.scala
package fr.in2p3.jsaga.adaptor.batchssh.job

import org.ogf.saga.error.NoSuccessException
import fr.in2p3.jsaga.adaptor.job.monitor.JobStatus
import fr.in2p3.jsaga.adaptor.job.SubState

class BatchSSHJobStatus(jobId: String, state: Object, retCode: Int) extends JobStatus(jobId, state, state.toString) {

    def getModel = "pbs-ssh"
    
    def getSubState = {
      m_nativeStateCode match {
        case "C" | "E" =>
          retCode match {
            case 0 => SubState.DONE
            case 153 => SubState.DONE
            case 271 => SubState.CANCELED
            case _ => SubState.FAILED_ERROR
          }
        case "H" => SubState.SUSPENDED_ACTIVE
        case "Q" => SubState.RUNNING_QUEUED
        case "S" | "W" => SubState.SUSPENDED_QUEUED
        case "R" => SubState.RUNNING_ACTIVE
        case _ => throw new NoSuccessException("Unreconized state "+ m_nativeStateCode)
      }

    }
}
